### DISCUSSION TITLE
WSGIMiddleware is deprecated and will be removed in a future release

### DISCUSSION THREAD

**Discussion by wolfskaempf** [fastapi/fastapi#8404] [Questions] [answered]
### First Check

- [X] I added a very descriptive title to this issue.
- [X] I used the GitHub search to find a similar issue and didn't find it.
- [X] I searched the FastAPI documentation, with the integrated search.
- [X] I already searched in Google "How to X in FastAPI" and didn't find any information.
- [X] I already read and followed all the tutorial in the docs and didn't find an answer.
- [X] I already checked if it is not related to FastAPI but to [Pydantic](https://github.com/samuelcolvin/pydantic).
- [X] I already checked if it is not related to FastAPI but to [Swagger UI](https://github.com/swagger-api/swagger-ui).
- [X] I already checked if it is not related to FastAPI but to [ReDoc](https://github.com/Redocly/redoc).

### Commit to Help

- [X] I commit to help with one of those options 👆

### Example Code

```python
from fastapi import FastAPI
from fastapi.middleware.wsgi import WSGIMiddleware
from flask import Flask, escape, request

flask_app = Flask(__name__)


@flask_app.route("/")
def flask_main():
    name = request.args.get("name", "World")
    return f"Hello, {escape(name)} from Flask!"


app = FastAPI()


@app.get("/v2")
def read_main():
    return {"message": "Hello World"}


app.mount("/v1", WSGIMiddleware(flask_app))
```

### Description

* `fastapi.middleware.wsgi` is inherited from Starlette
* Starlette will remove `starlette.middleware.wsgi` in the future ([See here](https://github.com/encode/starlette/blob/master/starlette/middleware/wsgi.py#L11))
* `fastapi.middleware.wsgi` also has the deprecation and removal warning due to its import of  `WSGIMiddleware` ([See here](https://github.com/tiangolo/fastapi/blob/master/fastapi/middleware/wsgi.py#L1))
* It is unclear if FastAPI will continue to support `WSGIMiddleware` from `fastapi.middleware.wsgi` or whether it too will be removed in the future

How to solve
* Please clarify if `fastapi.middleware.wsgi` and / or `WSGIMiddleware` will continue to be supported or if it will be removed when Starlette removes it (which it will if no further action is taken)
* Update docs to reflect an upcoming removal (if it will be removed)
* Suppress deprecation warning from `starlette.middleware.wsgi` (if `wsgi.WSGIMiddleware` will not be removed from FastAPI)

* [x] I commit to helping implement these changes.

### Operating System

macOS

### FastAPI Version

0.85.2

### Python Version

Python 3.11.0

### Additional Context

Thank you everyone for your hard work!

**Comment by TheJumpyWizard** [comment]
Opinion on https://github.com/abersheeran/a2wsgi? @tiangolo It's referenced in Starlette docs and is actively maintained

**Comment by Kludex** [comment, marked as answer]
Use https://github.com/abersheeran/a2wsgi. What's there to question about? @abersheeran knows far more about that than anyone else in the ecosystem. 🤷‍♂️

**Reply by abersheeran** [reply]
Thanks 🙏

**Reply by sbor23** [reply]
It's still part of the documentation, should the WSGIMiddleware be deprecated and removed as well in the future? Or will it just be dropped once starlette removes it?
https://fastapi.tiangolo.com/advanced/wsgi/
